ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/SchemaCache.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.schemas;
+
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.schemas.SchemaTypes.SchemaId;
+
+public interface SchemaCache {
+  static SchemaCache createDefault() {
+    return new SchemaCache() {
+      private final Map<SpecMilestone, Map<SchemaId<?>, Object>> cache =
+          new EnumMap<>(SpecMilestone.class);
+
+      @SuppressWarnings("unchecked")
+      @Override
+      public <T> T get(final SpecMilestone milestone, final SchemaId<T> schemaId) {
+        return (T) cache.computeIfAbsent(milestone, __ -> new HashMap<>()).get(schemaId);
```

Comment:
I am a bit confused by this part. If we are always priming the registry, and the registry is using `SchemaCache.put()` to update the cache, why do we need to have the `cache.computeIfAbsent` logic here? Can't we just use `get()`?

I belive if we use `computeIfAbsent` here, we will risk updating the underlying EnumMap concurrently, unless I am missing something?

Reply:
you are right..traces of me using AI here...